### PR TITLE
Pruning with restricted bits analysis

### PR DESCRIPTION
**Pull request by manasij7479**

### CODE REVIEW

**Comment by manasij7479** [comment]
Also converts a bunch of existing tabs to spaces.

**Comment by regehr** [comment]
the tabs and spaces need to be in a separate pull req

**Comment by regehr** [comment]
is the usePartialEval flag really needed? default values for arguments make things a lot more confusing

File: unittests/Interpreter/InterpreterInfra.cpp [hunk 1]
```diff
@@ -525,3 +524,121 @@ bool CRTesting::testFn(Inst::Kind pred) {
 
   return true;
 }
+
+bool RBTesting::nextRB(llvm::APInt& Val) {
+  if (Val.isAllOnesValue()) {
+    return false;
+  } else {
+    Val = Val + 1;
```
Comment (regehr) [inline unittests/Interpreter/InterpreterInfra.cpp]: ++Val;

File: unittests/Interpreter/InterpreterInfra.cpp [hunk 2]
```diff
@@ -525,3 +524,121 @@ bool CRTesting::testFn(Inst::Kind pred) {
...
+// Probably refactor to unify these two
+std::vector<llvm::APInt> explodeRestrictedBits(llvm::APInt X) {
+  std::vector<llvm::APInt> Result;
+  Result.push_back(llvm::APInt(WIDTH, 0));
+
+  for (int i = 0; i < WIDTH; ++i) {
+    if ( (X & (1 << i)) != 0 ) {
+      auto Copy = Result;
+      for (auto &Y : Copy) {
+        Result.push_back(Y | 1 << i);
```
Comment (regehr) [inline unittests/Interpreter/InterpreterInfra.cpp]: parenthesize to indicate precedence, as you've done elsewhere

**Comment by regehr** [comment]
LGTM thanks

**Pull request merged**